databroker/eventsource/archiver.py

```diff
@@ -73,65 +119,175 @@ def descriptors_given_header(self, header):
                         'external_url': self.url}
                 descs.append(desc)
             self._descriptors[run_start_uid] = descs
-            return [ d for d in self._descriptors]
+            
+            return self._descriptors[run_start_uid]      
```

Comment:
We probably want to return `list(self._descriptorts[run_start_uid)` because lists are mutable and if the user mutates the return value we want to be protected from that.